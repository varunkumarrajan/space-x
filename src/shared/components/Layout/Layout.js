// React
import React from 'react';
// Prop types
import PropTypes from 'prop-types';
// Styled component
import styled from 'styled-components';
// Components
import Header from './Header/Header';
import Footer from './Footer/Footer';
// Device
import device from 'config/BreakPoints';

// Styled main wrapper
const MainWrapper = styled.main`
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  @media screen and ${device.mobile} {
    flex-flow: row wrap;
  }
`;

/**
 * @description Layout
 * @param {object} { children, content: { header, footer} }
 */
const Layout = ({ children, content: { header, footer } }) => (
  <React.Fragment>
    <Header content={header} />
    <MainWrapper data-testid="mainId">{children}</MainWrapper>
    <Footer content={footer} />
  </React.Fragment>
);

// Default layout
export default Layout;

// Default props
Layout.defaultProps = {
  children: '',
  content: {
    header: {},
    footer: {},
  },
};

// Prop types
Layout.propTypes = {
  children: PropTypes.any.isRequired,
  content: PropTypes.shape({
    header: PropTypes.object,
    footer: PropTypes.object,
  }).isRequired,
};
