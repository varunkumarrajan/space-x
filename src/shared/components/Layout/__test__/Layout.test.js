// React
import React from 'react';
// Testing library
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Layout from './../Layout';

// Mock data
const layoutMock = {
  children: '',
  content: {
    header: {},
    footer: {},
  },
}

afterEach(cleanup);

describe('Layout', () => {
  it('Should Render Layout with props', () => {
    const { container, getByTestId } = render(<Layout content={layoutMock} />); 
    const {className} = getByTestId('mainId');
    expect(className).toBeTruthy();
    expect(container.children).not.toHaveLength(0);
  });

  it('Should Layout Snapshot', () => {
    const layout = renderer.create(<Layout content={layoutMock} />).toJSON();
    expect(layout).toMatchSnapshot();
  });
});
