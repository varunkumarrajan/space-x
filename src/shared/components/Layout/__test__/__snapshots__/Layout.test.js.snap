// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Should Layout Snapshot 1`] = `
Array [
  <header
    className="sc-bdfBwQ jIqIGL"
    data-testid="headerId"
  >
    
  </header>,
  <main
    className="sc-hKgILt bCXTwy"
    data-testid="mainId"
  >
    
  </main>,
  <footer
    className="sc-gsTCUz eZedek"
    data-testid="footerId"
  >
    
    : 
    <div
      className="sc-dlfnbm ieUdFR"
      data-testid="footerChildId"
    >
      
    </div>
  </footer>,
]
`;
